builder.md

```diff
@@ -1,58 +1,119 @@
 # Builder Guidelines
 **Version 0.10.1**
 
-A base16 builder is an application that can build syntax highlighting definition files for text editors by using base16 scheme files which contain a collection of colours and base16 template files which contain syntax highlighting rules.
+*The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).*
 
-Builders are designed for theme maintainers' ease of use. Theme maintainers should provide built versions of their theme so the end user doesn't need to be aware of the builder.
+A base16 builder is an application that builds application specific themeing configurations. It does this by using base16 scheme files (containing a collection of colors) and base16 template files (instructions concerning how to build the application specific files).
 
-## File Structure
+Builders are generally designed for template maintainers' ease of use. Template maintainers SHOULD provide built versions of their template so the end user doesn't need to be aware of the builder.
+
+## Definitions
+
+A _base16 scheme_ is a YAML file that represents a palette of 16 colors. For
+example: the [solarized
+scheme](https://github.com/base16-project/base16-schemes/blob/main/solarized-dark.yaml)
+
+A _base16 template_ is a mustache file that acts as a blueprint; it represents
+how to translate the scheme into an application's desired format. For example: the [vim
+template](https://github.com/base16-project/base16-vim/blob/main/templates/default.mustache)
+is used to convert a _base16 scheme_ into a vim colorscheme.
+
+A _base16 builder_ is an application that implements the full _building_
+feature specification, but MAY include additional functionality. These are
+usually targeted at template maintainers or when building other base16 tooling.
+
+_Building_ a _template_ is the process of replacing its variables (section 4.1)
+with ones extracted from a _scheme_; usually outputting it to a file, as
+defined by the _template_.
+
+## Inputs
 
 ### Schemes Repository
 
-The schemes repo should either be stored in a common location (perhaps referred to by environment variable or command line flag) or dynamically embedded in the builder.
+Convenient access to schemes in the [schemes repository](https://github.com/base16-project/base16-schemes) SHOULD be provided. The builder SHOULD store the repo in a common location (perhaps referred to by environment variable or command line flag), dynamically embed the schemes in the builder, or provide an alternate method of accessing the schemes.
```

Comment:
"should store"

I think this makes a builder sound like a manager - but we could talk about this stuff next time around... The "builder must keep track of schemes" bit still bugs me a bit... 

I think overall the changes here are looking great though.  I think perfecting this without a truly collaborative editor is going to take a few passes.

> "the schemes repo"

Are we purposely forbidding secondary themes repositories?

Reply:
I'm not sure the spec needs to talk about schemes in aggregate at all.  If we do perhaps we should refer to a "schemes directory"... the fact that someone got that via zip or tar.gz or git shouldn't matter.

Reply:
Alright, I've cleaned it up to make it clear that loading schemes is the only requirement - additionally loading from a directory or the GH repo directly have been changed to "MAY".